atomic_reactor/util.py

```diff
@@ -1125,3 +1125,14 @@ def get_primary_images(workflow):
             ImageName.parse(primary) for primary in
             workflow.build_result.annotations['repositories']['primary']]
     return primary_images
+
+
+def split_module_spec(module):
+    pieces = module.split('-')
```

Comment:
The first portion of the module spec can actually contain dashes, right?
We should use rsplit, but! I don't think that's safe either... (my-module-3.0 will be incorrectly parsed as name=my, version=module, stream=3.0)

ODCS has recently changed to use ":" separators instead (https://pagure.io/odcs/pull-request/157). But that's not deployed yet. How about if we

1. If `module` contains `":"`, use that as the splitter
2. Otherwise, splitter is "-", but we log a warning that "module name with dashes may be incorrectly parsed"

Wdyt?

Reply:
Actually, we should combine what's already done in `_resolve_modules` by regex. We can leave the change from `"-"` to another story `":"`.

Reply:
ok will use that regex